components/brave_shields/ios/browser/brave_shields_utils.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_BRAVE_SHIELDS_IOS_BROWSER_BRAVE_SHIELDS_UTILS_H_
+#define BRAVE_COMPONENTS_BRAVE_SHIELDS_IOS_BROWSER_BRAVE_SHIELDS_UTILS_H_
+
+#ifdef __cplusplus
+#include "brave/components/brave_shields/ios/common/brave_shields_panel.mojom.objc.h"
+#else
+#import "brave_shields_panel.mojom.objc.h"
+#endif
+
+@protocol BraveShieldsUtils
+
+- (BOOL)braveShieldsEnabledFor:(NSURL*)url
+    NS_SWIFT_NAME(isBraveShieldsEnabled(for:));
+- (void)setBraveShieldsEnabled:(BOOL)isEnabled forURL:(NSURL*)url;
+
+@property(nonatomic) BraveShieldsAdBlockMode defaultAdBlockMode;
```

Comment:
the api should be consistent on all platforms so we should either remove this or use it on all platforms instead of passing an empty GURL

Reply:
also seems odd to have it for adblock mode, but not the other settings. I would leave this out and we can discuss it maybe for a follow up